pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTLiteralExpression.java

```diff
@@ -25,64 +26,71 @@ public Object jjtAccept(ApexParserVisitor visitor, Object data) {
         return visitor.visit(this, data);
     }
 
-    /*
+    public enum LiteralType {
+        STRING,
+        INTEGER,
+        LONG,
+        DOUBLE,
+        DECIMAL,
+        TRUE,
+        FALSE,
+        NULL
+    }
+
     public LiteralType getLiteralType() {
-        return node.getLiteralType();
+        if (node instanceof LiteralExpression.StringVal) {
+            return LiteralType.STRING;
+        } else if (node instanceof LiteralExpression.BooleanVal) {
+            return ((LiteralExpression.BooleanVal) node).getValue() ? LiteralType.TRUE : LiteralType.FALSE;
+        } else if (node instanceof LiteralExpression.IntegerVal) {
+            return LiteralType.INTEGER;
+        } else if (node instanceof LiteralExpression.DoubleVal) {
+            return LiteralType.DOUBLE;
+        } else if (node instanceof LiteralExpression.LongVal) {
+            return LiteralType.LONG;
+        } else if (false) { // TODO(b/239648780)
```

Comment:
just curious as to why this is the only not-implemented (presumably "to be implemented later"), since I would expect that LONG is just about identical to INTEGER...

Reply:
It is `DECIMAL` that is unimplemented, not `LONG`. Summit currently has no concept of a `Decimal` literal.